app/models/medium.rb

```diff
@@ -295,11 +304,12 @@ def self.search_by(search_params, page)
     if search_params[:all_tags] == '1' && search_params[:tag_operator] == 'and'
       search_params[:tag_ids] = Tag.pluck(:id)
     end
-    admin = User.find_by(id: search_params[:user_id])&.admin?
+    user = User.find_by(id: search_params[:user_id])
```

Comment:
So I can pretend I'm another user by sending the corresponding user id in the search params? Where is this mysterious `search_params` variable declared? Is this provided by Ruby (on Rails) by parsing the URL? I cannot find it in the documentation. Is it sanitized?

Reply:
The search forms route to the media_controller, you can see this in this line

```
form_with url: media_search_path
```

in the media search partials. In the media controller, there is a search_params method which does filtering ([strong parameters](https://edgeguides.rubyonrails.org/action_controller_overview.html#strong-parameters) as it is called in Rails). In this method, `search_params[:user_id]` is set via

```
      params[:search][:user_id] = current_user.id
```
so you should be able to alter that as a user. In the above mentioned service model I suggested, one could pass a user (the current user) for initialization.

Reply:
Ah ok, that's the reason I couldn't find a `search_params = ` statement in the codebase (except for some tests). But where is `params[:search]` mapped to the variable `search_params`? Is this done internally by rails? I guess `search_params[:user_id]` is performing a call to the `search_params()` method in the media controller, getting back and object and accessing the key `user_id` there. But why don't we have to use `search_params[:search][:user_id]` then?

Reply:
I'll explain it in our meeting, okay?